lib/portlayer/storage/vsphere/volume.go

```diff
@@ -189,8 +203,26 @@ func (v *VolumeStore) VolumeCreate(ctx context.Context, ID string, store *url.UR
 	return vol, nil
 }
 
-func (v *VolumeStore) VolumeDestroy(ctx context.Context, ID string) error {
-	return fmt.Errorf("TBD.  Not supported yet")
+func (v *VolumeStore) VolumeDestroy(ctx context.Context, vol *storage.Volume) error {
+	if err := volumesInUse(ctx, vol.ID); err != nil {
```

Comment:
You're implying compare and swap.  No, there is no guarantee that the image you're going to RM below is not in use after exec tells us what images there are because the exec has no lock to speak of.  But, you're the authority here.  Is there a lock we can hold?

However, popping up a level, vsphere will keep things consistent for us in any case.  If we iterate the images in use, find that the image we're going to RM is not used, then RM the image as a new container is spinning up,  vsphere will either return an error on the container creation path, or prevent the image from being removed because the container is running (and return an error to the caller of `rm`).

In my view, that's sufficient.  Thoughts @hickeng ?